e2e/benchmarks/services_test.go

```diff
@@ -0,0 +1,276 @@
+// +build e2e
+
+// Runs benchmarks using the Null provider and working with Consul setup with
+// multiple Services or Service Instances.
+
+package benchmarks
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/command"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const configFile = "config.hcl"
+const tempDirPrefix = "tmp_"
+
+// Benchmark N Services Instance (all the same service)
+func Benchmark1Instance(b *testing.B) {
```

Comment:
Minor suggestion: To make this regex friendly, structure the function names `BenchmarkInstances1XXX` allows for you to selectively run a set of benchmarks

```
go test ./e2e/benchmarks -bench=BenchmarkInstance
```

would run all 3 and not run the other benchmarks in `tasks_test.go` etc.

Reply:
Yeah... I went the other way. Made it so it was easy to call one without the others via regex. 
Eg.

`go test -tags e2e -bench 1I` 

Where `1I` can be `100I`, `1000I` or `1S` etc. 
Then for more you use `*`, eg. `1*I` or `1*S`.

I'm not married to the idea. It was just the most useful as I was developing them. If the other way is preferred, I'll change it no worries.

Reply:
Oh! TIL that it works the other way. Go continues to surprise me, thanks for sharing.